framework/test/gtest/BundleTrackerCustomCallbackTest.cpp

```diff
@@ -0,0 +1,661 @@
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  Bundle bundle2 = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleM");
+
+  // bundle: INSTALLED ---> uninstalled (Remove)
+  EXPECT_CALL(*customizer, AddingBundle).Times(0);
+  EXPECT_CALL(*customizer, ModifiedBundle).Times(0);
+  EXPECT_CALL(*customizer, RemovedBundle).Times(1);
+  bundle.Uninstall();
+
+  // When the BundleTracker is closed, RemovedBundle should be called
+  // for the tracked bundle bundle2
+  EXPECT_CALL(*customizer, RemovedBundle).Times(::testing::AnyNumber());
+  EXPECT_CALL(*customizer,
+              RemovedBundle(HasName("TestBundleM"), ::testing::_, ::testing::_))
+    .Times(1);
+
+  bundleTracker.Close();
+}
+
+//
+// Test custom implementation through a BundleTrackerCustomizer subclass tracking custom objects
+//
+class MockCustomizerWithObject : public BundleTrackerCustomizer<int>
+{
+public:
+  MOCK_METHOD(std::optional<int>,
+              AddingBundle,
+              (const Bundle&, const BundleEvent&),
+              (override));
+  MOCK_METHOD(void,
+              ModifiedBundle,
+              (const Bundle&, const BundleEvent&, int),
+              (override));
+  MOCK_METHOD(void,
+              RemovedBundle,
+              (const Bundle&, const BundleEvent&, int),
+              (override));
+};
+
+TEST_F(BundleTrackerCustomCallbackTest,
+       ObjectIsTrackedWhenReturnedByAddingBundleFromCustomizer)
+{
+  auto stateMask =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_ACTIVE);
+  auto customizer = std::make_shared<MockCustomizerWithObject>();
+  auto bundleTracker = BundleTracker<int>(context, stateMask, customizer);
+
+  // Make AddingBundle return null for bundles we aren't testing (main, system_bundle)
+  EXPECT_CALL(*customizer, AddingBundle)
+    .WillRepeatedly(::testing::Return(std::nullopt));
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+
+  // When bundle makes the following state transitions:
+  // ---> installed ---> resolved ---> starting ---> ACTIVE (Add)
+  // Then there should be 1 AddingBundle callback, and
+  // When it returns the object 5
+  EXPECT_CALL(*customizer, AddingBundle)
+    .Times(1)
+    .WillOnce(::testing::Return(5)); // returns the object 1
+  EXPECT_CALL(*customizer, ModifiedBundle).Times(0);
+  EXPECT_CALL(*customizer, RemovedBundle).Times(0);
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  bundle.Start();
+
+  // Then the object 5 should be the tracked
+  EXPECT_EQ(5, bundleTracker.GetObject(bundle))
+    << "The object returned by a customizer's AddingBundle should be tracked";
+
+  // Whatever happens after the bundle is added is out of scope for this test
+  EXPECT_CALL(*customizer, RemovedBundle).Times(::testing::AnyNumber());
```

Comment:
I see a number of these `EXPECT_CALL`s throughout these tests that follow the pattern of:
1. have a comment of the form: "Whatever happens after <XYZ> is out of scope for this test"
2. have `EXPECT_CALL(*customizer, RemovedBundle).Times(::testing::AnyNumber());`

If the behavior is out of scope, an EXPECT_CALL isn't necessary. What am I missing?